tests/cnf/core/network/metallb/tests/bgp-connect-time-tests.go

```diff
@@ -0,0 +1,224 @@
+
+		By("Listing master nodes")
+		masterNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.ControlPlaneLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Fail to list master nodes")
+		Expect(len(masterNodeList)).To(BeNumerically(">", 0),
+			"Failed to detect master nodes")
+	})
+
+	BeforeEach(func() {
+		By("Creating a new instance of MetalLB Speakers on workers")
+		err := metallbenv.CreateNewMetalLbDaemonSetAndWaitUntilItsRunning(tsparams.DefaultTimeout, workerLabelMap)
+		Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+
+		_, err = pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+			LabelSelector: tsparams.FRRK8sDefaultLabel,
+		})
+		Expect(err).ToNot(HaveOccurred(), "Fail to list speaker pods")
+
+		By("Collecting information before test")
+		frrk8sPods, err = pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+			LabelSelector: tsparams.FRRK8sDefaultLabel,
+		})
+		Expect(err).ToNot(HaveOccurred(), "Failed to list speaker pods")
+
+	})
+
+	AfterEach(func() {
+		By("Cleaning MetalLb operator namespace")
+		metalLbNs, err := namespace.Pull(APIClient, NetConfig.MlbOperatorNamespace)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull metalLb operator namespace")
+		err = metalLbNs.CleanObjects(
+			tsparams.DefaultTimeout,
+			metallb.GetBGPPeerGVR(),
+			metallb.GetBFDProfileGVR(),
+			metallb.GetBGPPeerGVR(),
+			metallb.GetBGPAdvertisementGVR(),
+			metallb.GetIPAddressPoolGVR(),
+			metallb.GetMetalLbIoGVR())
+		Expect(err).ToNot(HaveOccurred(), "Failed to remove object's from operator namespace")
+
+		By("Cleaning test namespace")
+		err = namespace.NewBuilder(APIClient, tsparams.TestNamespaceName).CleanObjects(
+			tsparams.DefaultTimeout,
+			pod.GetGVR(),
+			service.GetServiceGVR(),
+			configmap.GetGVR(),
+			nad.GetGVR())
+		Expect(err).ToNot(HaveOccurred(), "Failed to clean test namespace")
+	})
+
+	It("Verify configuration of a FRR node router peer with the connectTime less than the default of 120 seconds",
+		reportxml.ID("74414"), func() {
+
+			By("Creating BGP Peers with 10 second retry connect timer")
+			createBGPPeerWithConnectTimeAndVerifyIfItsReady(metav1.Duration{Duration: 10 * time.Second},
+				ipv4metalLbIPList[0], "", 64500, false, frrk8sPods)
+
+			By("Validate BGP Peers with 10 second retry connect timer")
+			Eventually(func() string {
+				// Get the routes
+				connectTime, err := frr.VerifyBGPConnectTime(frrk8sPods, ipv4metalLbIPList[0], 10)
+				Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP connect time")
+
+				// Return the routes to be checked
+				return connectTime
+			}, 60*time.Second, 5*time.Second).Should(ContainSubstring("ConnectRetryTimer: 10"),
+				"Fail to find received BGP connect time")
+		})
+
+	It("Verify the retry timers connect on an established Metallb neighbor with a timer connect less "+
```

Comment:
not sure I understand the different between the former tc, can't they be merged together?

Reply:
Yes the second test actually includes both the first and the second. But they are separate test cases in polarion. I can talk to Evgeny about it